File: tuan07/bai1_ca_nhan.py
# Sinh ke tiep; moi cau hinh duoc in va dem mot lan.
def liet_ke(n: int) -> int:
    x = [0] * n
    dem = 0
    k = 1  # Tham so ca s = 6
    dem_dat = 0

    while True:
        # Chi in va dem cac xau co dung k bit 1
        if sum(x) == k:
            print("".join(map(str, x)))
            dem_dat += 1

        dem += 1
        i = n - 1
        while i >= 0 and x[i] == 1:
            x[i] = 0  # Xoa cac bit 1 lien tiep o duoi.
            i -= 1
        if i < 0:
            break
        x[i] = 1

    print("So xau dat:", dem_dat)
    return dem


def main():
    print("MSSV: N24DCDT067 | Ma ca: 6")
    print("Nhap n (1..10):")
    try:
        n = int(input().strip())
    except (ValueError, EOFError):
        print("Du lieu khong hop le.")
        return
    if not 1 <= n <= 10:
        print("Du lieu khong hop le.")
        return
    dem = liet_ke(n)
    print("Tong so xau da duyet:", dem)


if __name__ == "__main__":
    main()

File: tuan07/bai1_ca_nhan.jl
# Julia danh chi so mang tu 1.
function liet_ke(n::Int)
    x = zeros(Int, n)
    dem = 0
    k = 1  # Tham so ca s = 6
    dem_dat = 0

    while true
        # Chi in va dem cac xau co dung k bit 1
        if sum(x) == k
            println(join(x))
            dem_dat += 1
        end

        dem += 1
        i = n
        while i >= 1 && x[i] == 1
            x[i] = 0  # Xoa cac bit 1 lien tiep o duoi.
            i -= 1
        end
        if i < 1
            break
        end
        x[i] = 1
    end

    println("So xau dat: ", dem_dat)
    return dem
end

function main()
    println("MSSV: N24DCDT067 | Ma ca: 6")
    println("Nhap n (1..10):")
    n = tryparse(Int, strip(readline()))
    if n === nothing || !(1 <= n <= 10)
        println("Du lieu khong hop le.")
        return
    end
    dem = liet_ke(n)
    println("Tong so xau da duyet: ", dem)
end

main()